roles/st2mistral/tasks/main.yml

```diff
@@ -55,9 +42,29 @@
     - restart mistral
   tags: st2mistral
 
-- name: Setup Mistral DB tables, etc
+- name: Make sure "Initiate mistral database" doesn't run twice
   become: yes
-  shell: /opt/stackstorm/mistral/bin/mistral-db-manage --config-file /etc/mistral/mistral.conf upgrade head && touch /etc/mistral/init_mistral_db.SQL.ansible.has.run
+  file:
+    path: /etc/mistral/init_mistral_db.SQL.ansible.has.run
+    state: touch
+  when: mistral_initiate_database.changed
+  tags: st2mistral, skip_ansible_lint
+
+- name: Configure mistral
+  become: yes
+  ini_file:
+    dest: /etc/mistral/mistral.conf
+    section: database
+    option: connection
+    value: postgresql://{{ st2mistral_db_username }}:{{ st2mistral_db_password }}@localhost/{{ st2mistral_db }}
+    backup: yes
+  notify:
+    - restart mistral
+  tags: st2mistral
+
+- name: Setup mistral DB tables, etc
```

Comment:
Same as above - don't currently see how this is idempotent, since there is no `when` clause